tests/unit/test_base.py

```diff
@@ -90,6 +90,30 @@ def test_instantiate_non_dict(self, fake_gitlab, fake_manager):
         with pytest.raises(gitlab.exceptions.GitlabParsingError):
             FakeObject(fake_manager, ["a", "list", "fails"])
 
+    def test_missing_attribute(self, fake_gitlab, fake_manager):
+        obj = FakeObject(manager=fake_manager, attrs={"foo": "bar"})
+        with pytest.raises(AttributeError) as excinfo:
+            obj.missing_attribute
+        assert str(excinfo.value) == "missing_attribute"
+        assert "was created via a list()" not in str(excinfo.value)
+        assert (
+            "https://python-gitlab.readthedocs.io/en/stable/faq.html"
+            "#attribute-error-list"
+        ) not in str(excinfo.value)
+
+        # Test for more informative message if a list() created object
+        obj = FakeObject(
+            manager=fake_manager, attrs={"foo": "bar"}, list_created_object=True
+        )
+        with pytest.raises(AttributeError) as excinfo:
+            obj.missing_attribute
+        assert "missing_attribute" in str(excinfo.value)
+        assert "was created via a list()" in str(excinfo.value)
+        assert (
+            "https://python-gitlab.readthedocs.io/en/stable/faq.html"
+            "#attribute-error-list"
+        ) in str(excinfo.value)
+
```

Comment:
IMO if we split this into 2 exact tests we make sure we hit all asserts and can skip some comments.

```suggestion
    def test_missing_attribute_does_not_raise_custom(self, fake_gitlab, fake_manager):
        obj = FakeObject(manager=fake_manager, attrs={"foo": "bar"})
        with pytest.raises(AttributeError) as excinfo:
            obj.missing_attribute
        assert str(excinfo.value) == "missing_attribute"
        assert "was created via a list()" not in str(excinfo.value)

    def test_missing_attribute_from_list_raises_custom(self, fake_gitlab, fake_manager):
        obj = FakeObject(
            manager=fake_manager, attrs={"foo": "bar"}, list_created_object=True
        )
        with pytest.raises(AttributeError) as excinfo:
            obj.missing_attribute
        assert "missing_attribute" in str(excinfo.value)
        assert "was created via a list()" in str(excinfo.value)
```

Also should we omit the URL in case this gets moved around? I did in the suggestion. Otherwise maybe some constant/template reused from production code might help.

Reply:
I like that idea to split up the tests. I will do that.

I would like to keep the URL part. I use this program called `shellcheck` and one of the things I love about it is that it gives me URLs for errors. Now maybe we figure out a best URL to put stuff, but having the URL so someone can just click it to find information on how to resolve their issue is a big win in my opinion.

An example of what it does:
```
In foo.sh line 5:
echo $1
     ^-- SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
echo "$1"

For more information:
  https://www.shellcheck.net/wiki/SC2086 -- Double quote to prevent globbing ...
```

For a first step let me put the template/URL of the message into the code and that way not duplicating as much.

Reply:
Ah yes sorry I just meant in the test. I'd definitely keep it in the exception message, it's really helpful. So if it's templated that works for me!

Reply:
Done